test/smart-contract-client.test.js

```diff
@@ -0,0 +1,101 @@
+import { assertEquals, assertRejects, assert } from 'zinnia:assert'
+import { getMinerPeerIdFromSmartContract } from '../lib/smart-contract-client.js'
+import { getMinerPeerId } from '../lib/miner-info.js'
+import { test } from 'zinnia:test'
+
+const mockPeerIdResponse = {
+  peerID: '12D3KooWGQmdpbssrYHWFTwwbKmKL3i54EJC9j7RRNb47U9jUv1U',
+  signature: '0x1234567890abcdef'
+}
+
+const mockEmptyPeerIdResponse = {
+  peerID: '',
+  signature: '0x'
+}
+
+// Mock contract factory
+function createMockContract (mockResponses) {
+  return {
+    getPeerData: async (minerId) => {
+      const response = mockResponses[minerId]
+      if (!response) {
+        throw new Error(`Miner ID ${minerId} not found in contract`)
+      }
+      return response
+    }
+  }
+}
+
+test('getMinerPeerIdFromSmartContract returns peer ID for valid miner ID', async () => {
+  // Create mock contract with predefined responses
+  const peerId = 12345
+  const mockContract = createMockContract({
+    [peerId]: mockPeerIdResponse
+  })
+
+  const actualPeerId = await getMinerPeerIdFromSmartContract(`f0${peerId}`, {
+    smartContract: mockContract
+  })
+
+  assertEquals(actualPeerId, mockPeerIdResponse.peerID)
+})
+
+test('getMinerPeerIdFromSmartContract returns correct peer id for miner f03303347', async () => {
+  const peerId = await getMinerPeerIdFromSmartContract('f03303347')
+  assertEquals(typeof peerId, 'string', 'Expected peerId to be a string')
+  assertEquals(peerId, '12D3KooWJ91c6xQshrNe7QAXPFAaeRrHWq2UrgXGPf8UmMZMwyZ5')
+})
+
+test('getMinerPeerIdFromSmartContract returns empty string for miner ID with no peer ID', async () => {
```

Comment:
Could you please add a code comment to capture the first part of https://github.com/CheckerNetwork/spark-checker/pull/117#discussion_r1987651113?

> The contract specifies this behaviour here:
> 
> ```solidity
>         if (bytes(data.peerID).length == 0) {
>             return PeerData("", bytes(""));
>         }
> ```

Reply:
> Could you please add a code comment to capture the first part of [#117 (comment)](https://github.com/CheckerNetwork/spark-checker/pull/117#discussion_r1987651113)?
> 
> > The contract specifies this behaviour here:
> > ```solidity
> >         if (bytes(data.peerID).length == 0) {
> >             return PeerData("", bytes(""));
> >         }
> > ```

@NikolasHaimerl PTAL ☝🏻